shed/event_streams.py

```diff
@@ -760,15 +772,23 @@ class accumulate(EventStream):
     def __init__(self, func, child, state_key=None,
                  full_event=False,
                  output_info=None,
-                 input_info=None, start=no_default):
+                 input_info=None, start=no_default,
+                 across_start=False):
         self.state_key = state_key
         self.func = func
+        self.start_state = start
         self.state = start
         EventStream.__init__(self, child, input_info=input_info,
                              output_info=output_info)
         self.full_event = full_event
+        self.across_start = across_start
         self.generate_provenance(function=func)
 
+    def start(self, docs):
+        if not self.across_start:
```

Comment:
seems like something you could do in `__init__` to save resources (if possible)

```python
if self.across_start:
    self.start = self._start
```
etc.